core/lib/dal/src/tee_verifier_input_producer_dal.rs

```diff
@@ -72,9 +72,57 @@ impl TeeVerifierInputProducerDal<'_, '_> {
         Ok(())
     }
 
+    // Returns inclusive range, i.e. [low,high]
+    pub async fn get_new_l1_batches(
+        &mut self,
+    ) -> DalResult<Option<(L1BatchNumber, L1BatchNumber)>> {
+        // COALESCE(1, ...) covers the case where the table is initially empty. NB: We start from batch 1, since genesis batch 0 has no proof.
+        let row_low = sqlx::query!(
+            r#"
+            SELECT
+                COALESCE(1, MAX(l1_batch_number) + 1) AS "number"
+            FROM
+                tee_verifier_input_producer_jobs
+            "#
+        )
+        .instrument("get_latest_tee_job_l1_batch_number")
+        .report_latency()
+        .fetch_one(self.storage)
+        .await?;
+
+        // Since we depend on Merkle paths, we use the proof_generation_details table to inform us of newly available to-be-proven batches.
+        let row_high = sqlx::query!(
+            r#"
+            SELECT
+                MAX(l1_batch_number) AS "number"
+            FROM
+                proof_generation_details
```

Comment:
I'm not sure why `proof_gen_data_blob_url` is allowed to be `NULL` in the database, but in this case, shouldn't we ensure that Merkle paths are indeed available?
```suggestion
            SELECT
                MAX(l1_batch_number) AS "number"
            FROM
                proof_generation_details
            WHERE
                proof_gen_data_blob_url IS NOT NULL
```

Reply:
Just found that `proof_gen_data_blob_url` used to be non-`NULL`. The constraint was dropped recently here: https://github.com/matter-labs/zksync-era/pull/2405/files#diff-bf5de8ff2822841d4fd0dc5850c7a17039d594382e790b1c877b0949337fde37

Reply:
Yeah, makes sense.